Code/Tools/AssetProcessor/native/ui/MainWindow.cpp

```diff
@@ -2036,37 +2193,54 @@ void MainWindow::ShowSourceAssetContextMenu(const QPoint& pos)
                 {
                     return true;
                 }
-                ++productCount;
-                productAssetMenu.m_listWidget->addItem(productEntry.m_productName.c_str());
+                if (IsProductOutputFlagSet(productEntry, AssetBuilderSDK::ProductOutputFlags::IntermediateAsset))
+                {
+                    ++intermediateCount;
+                    intermediateAssetMenu.m_listWidget->addItem(AZStd::string(AssetUtilities::StripAssetPlatformNoCopy(productEntry.m_productName)).c_str());
+                }
+                else
+                {
+                    ++productCount;
+                    productAssetMenu.m_listWidget->addItem(productEntry.m_productName.c_str());
+                }
                 return true; // Keep iterating, add all products.
             });
-            return true; // Stop iterating, there should only be one job with this run key.
-        });
-        if (productCount == 0)
-        {
-            CreateDisabledProductAssetRightClickMenu(&menu, productAssetMenu.m_productMenu, tr("This source asset has no products."));
-            productAssetMenu.m_productMenu = nullptr;
-        }
-        else
-        {
-            ResizeProductAssetRightClickMenuList(productAssetMenu.m_listWidget, productCount);
-        }
+        return true; // Stop iterating, there should only be one job with this run key.
+    });
+    if (productCount == 0)
```

Comment:
There is some DRY violation here, not sure if it should be addressed here, but should be careful of code duplication.

Reply:
Cleaned it up a bit, but can you be specific about the code reuse here you don't like?